src/ol-layerswitcher.js

```diff
@@ -103,45 +103,45 @@ export default class LayerSwitcher extends Control {
     }
 
     /**
-    * Re-draw the layer panel to represent the current state of the layers.
+    * **Static** Re-draw the layer panel to represent the current state of the layers.
     */
-    renderPanel() {
+    static renderPanel(panel, map) {
 
-        this.ensureTopVisibleBaseLayerShown_();
+        LayerSwitcher.ensureTopVisibleBaseLayerShown_(map);
 
-        while(this.panel.firstChild) {
-            this.panel.removeChild(this.panel.firstChild);
+        while(panel.firstChild) {
+            panel.removeChild(panel.firstChild);
         }
 
         var ul = document.createElement('ul');
-        this.panel.appendChild(ul);
-        this.renderLayers_(this.getMap(), ul);
+        panel.appendChild(ul);
+        LayerSwitcher.renderLayers_(map, map, ul);
```

Comment:
This looks odd due to two `map` arguments :-)
Maybe worth add a comment to say that we're passing the map as the root of the layers tree.